### ISSUE TITLE
Switch away from Travis

### ISSUE THREAD

**Issue by TheLastProject** [Pext/Pext#214]
It has become a common occurrence to have to wait up to half an hour for Travis to even start building any builds. Travis sadly has only seemed to be getting worse and worse over time. It's time to consider alternatives.

**Comment by TheLastProject** [comment]
[Drone CI](https://drone.io/) seems to be the best possibility, seemingly having an Open Source "Community Edition". If we migrate to another platform, I'd rather it be an open one.

**Comment by TheAssassin** [comment]
I've recently switched almost all projects from Travis to GitHub actions, as [Travis CI doesn't want to be the number 1 service for open-source software any more](https://www.jeffgeerling.com/blog/2020/travis-cis-new-pricing-plan-threw-wrench-my-open-source-works). It works well, is fast, doesn't require personal access tokens for tasks like uploading releases (as there's always a project-specific token available), and is pretty much unlimited (similar to Travis a couple years ago).

Regarding Drone CI, indeed the server is licensed under the Apache 2 license (as well as multiple non-free licenses). However, the exec runners (and most notably, the Docker runner) is non-free. I've just set up Drone as a potential alternative to Jenkins (which I still like, but is very cumbersome to set up and maintain), and after being pretty much done found this issue. It's very sad to see that this "free software" can't be used under a true free software license (as compared to some "free for personal use" or so license).
TL;DR: Drone works (just a bit limited, as you cannot set up runners easily in real VMs), but being actually non-free, it's a big no-go.

In fact, I moved one of my projects to GitHub Actions because I didn't have the time to set up something on private infra: https://github.com/TheAssassin/prebuilt-cmake
It's uploading its binaries to one of my servers via WebDAV, though.

`<ad>` Oh, in case you are looking for a easy-to-use tool to upload stuff from Actions to GitHub, check out https://github.com/TheAssassin/pyuploadtool. `</ad>`

**Comment by TheLastProject** [comment]
Oops, we've been Travis-free for 2 years already it seems. Just need to update the build status in the README.

**Comment by TheLastProject** [comment]
Fixed in 2721d04aa7f4b9f58b213ae3709e1545b9925d9c.
